File: 14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if strs == [] or strs == ['']:
            return ''
        
        strs = sorted(strs, key = lambda x: len(x))

        def Check(match: str, i: int) -> bool:
            for s in strs:
                if not match == s[0:i]:
                    return False
            return True

        for i in range(len(strs[0])+1):
            if not Check(strs[0][0:i],i):
                return strs[0][0:i-1]
            
        return strs[0]

File: 14. Longest Common Prefix.rs

impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let mut answer: String = String::new();
        let min_length: usize = strs.iter().map(|s: &String| s.len()).min().unwrap_or(0);

        for i in 0..min_length{
            let first_char: Option<char> = strs[0].chars().nth(i);

            for s in &strs{
                if s.chars().nth(i) != first_char{
                    return answer;
                }
            }

            answer.push(first_char.unwrap());
        }
        
        return answer;
    }
}
